### ISSUE TITLE
Extraneous space on Flow interface method named `static`

### ISSUE THREAD

**Issue by wchargin** [prettier/prettier#4876]
**Prettier 1.13.7**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLWcBOAzAhmOAAgElDgAdKQwgZxlxlTAAoBKJQqAVwFsAjLAG5KAX0ogANCAgAHRtBrJQuTJggB3AAoqEilLgBuEVABNJIPpnwBrODADKM-OgDmyGJi5wp6GlhiaVi48uMh4ADZ+UgBWNAAeAEJWYLYOuDxwADLocGG4kd4gTph+mMgWuHwAnuHQ5jKY6DAA6qYwABbIABwADFINEH7NVjLlDXClBrlSmHAAjlyos4G4waFIEVEgfjyo7p6FNK7hcACKXBDweQVS9HytJh3IAEy3VqjhrgDCEDwh5VBoNMQFw-AAVSp6TZwEQiIA)
```sh
--parser babylon
```

**Input:**
```jsx
interface I {
  static(): number;
}

```

**Output:**
```jsx
interface I {
  static (): number;
}

```

**Expected behavior:**
```jsx
interface I {
  static(): number;
}

```

Note that this is perfectly valid, at both the type level and the value
level. The following code [typechecks with one expected error][1]…

```js
interface I {
  static(): number;
}
class Good implements I {
  static() {
    return 17;
  }
}
class Bad implements I {
  static() {
    // $ExpectError
    return "definitely not a number";
  }
}
```

[1]: https://flow.org/try/#0JYOwLgpgTgZghgYwgAgJLIN4ChnIM5hxjAIAUAlAFzIgCuAtgEbQDcWAvlggDZx57IA4gHthAE2TB6AB24R6EcAPTZcBIiQqYcuZFAhhaUEMgCMAdja5OnHnwEAhOBKmz5isMu1rCxMuW9dAHog5AASAFEAD2kIBDAIqChhKB1cfUNjZAAiMQgYUGBIbgBPGmEwZDgaBmYobKtkGyA

…and `new class { static() { return "hi" } }().static()` evaluates to
the string `"hi"` in Chrome 67 and Node v8.9.4.

h/t @decentralion

**Comment by bakkot** [comment]
This is a bug in babylon, looks like. Opened [upstream](https://github.com/babel/babel/issues/8373) issue.

**Comment by wchargin** [comment]
Agreed—I should have checked the AST. Thank you for the report and the PR to fix it!

**Comment by decentralion** [comment]
Thank you @bakkot! Also :wave:

**Comment by bakkot** [comment]
👋!
